nixos/modules/programs/command-not-found/rust/src/main.rs

```diff
@@ -0,0 +1,47 @@
+use std::env;
+use std::process::exit;
+use rusqlite::{params, Connection, Result};
+
+mod config;
+
+fn query_packages(system: &str, program: &str) -> Result<Vec<String>> {
+    let conn = Connection::open(config::DB_PATH)?;
+    let mut stmt = conn.prepare("select package from Programs where system = ? and name = ?;")?;
+    let iter = stmt.query_map(params![system, program], |row| { row.get(0) })?;
+    let mut pkgs: Vec<String> = vec![];
+    for pkg in iter {
+        pkgs.push(pkg?);
+    }
+    Ok(pkgs)
+}
+
+fn main() {
+    let args: Vec<_> = env::args().collect();
+    if args.len() < 2 {
+        eprintln!("USAGE: {} PROGRAM", args[0]);
+        exit(1);
```

Comment:
The documentation for [exit](https://doc.rust-lang.org/std/process/fn.exit.html) advises against calling `exit` in the middle of a programm, "If a clean shutdown is needed". I don't know if we do, but I'd rather err on the side of caution :sweat_smile:. The docs also give an example of patter to work around this. Maybe use that?

Reply:
This is called from the main function. It should be safe looking at what the program is doing currently. However I will put everything into a second function just to be safe against refactoring.